packages/react/src/components/Button/Button.tsx

```diff
@@ -69,7 +74,8 @@ export const Button = forwardRef<HTMLButtonElement, ButtonProps>(
           data-size={size}
           data-variant={variant}
           ref={ref}
-          type={type}
+          /* don't set type when we use `asChild` */
+          {...(asChild ? { asChild: true, type } : { type: 'button' })}
```

Comment:
```suggestion
          {...(asChild ? { asChild: true, type } : { type: type ?? 'button' })}
```
I guess we should do this as well :D

Reply:
NVM :D

Reply:
heh this is true, good catch 🗡️

Reply:
I'll add a test for setting type as well, since we are doing something to it, it would be nice to have a test for it